### PR TITLE
[docs] Minor revision of trailing closure matching educational note

### PR DESCRIPTION
**Pull request by xwu**
Belatedly coming to the educational note on trailing closure matching. This is a revision that principally reorganizes the information.

* The bit about the warning when forwards and backwards scans produce different results is pretty key, as it's likely to be the reason that users encounter this note; therefore, move that above the details about the forward scanning rule and insert a bit about how to silence this warning.

* Insert a brief summary about the forward scanning rule's detailed implementation upfront, and only for people that need the details, provide the nitty gritty definitions in their own sections afterwards.

### CODE REVIEW

**Comment by xwu** [comment]
@owenv Sorry that I couldn't get to reviewing #33119 before it was merged. Here are some changes I had in mind after looking through the document.

**Review by owenv — APPROVED** [review]
Looks great, thanks!

File: userdocs/diagnostics/trailing-closure-matching.md
Comment (3405691582) [inline userdocs/diagnostics/trailing-closure-matching.md:12]: s/animate:/animations:/, no?

**Review by xedin — APPROVED** [review]
LGTM! Once that typo in label, @3405691582 has noticed, is fixed everything should be good to go.

**Comment by xedin** [comment]
@xwu let me know if you need help with kicking off CI and merging...

File: userdocs/diagnostics/trailing-closure-matching.md
Comment (xwu) [reply userdocs/diagnostics/trailing-closure-matching.md:12]:
```suggestion
animate(withDuration: 0.3) /* `animations:` is unwritten. */ {
```

**Comment by xwu** [comment]
@swift-ci smoke test and merge

**Comment by xwu** [comment]
@swift-ci smoke test and merge

**Pull request merged**
